### ISSUE TITLE
Clarification on "GW projections"

### ISSUE THREAD

**Issue by AminBoumerdassi** [mikekatz04/lisa-on-gpu#8]
Hi,

I wanted to use Fast LISA response to propagate GW glitches through TDI. To generate the glitches I use the LISA glitch package which can simulate glitches injected at various points such as the test mass interferometer, inter-spacecraft interferometer etc.

From what I understand, it looks like I should be able to input the projections into the ```get_tdi_delays()``` method from the ```pyresponseTDI``` class but it's not clear to me what projections Fast LISA response accepts e.g. TMI, ISI etc.

Could you please clarify this?

-Amin

**Comment by j2bbayle** [comment]
Can I ask why you don't plan on using, say, PyTDI? Do you plan to do inference?

**Comment by AminBoumerdassi** [comment]
I haven't actually used PyTDI before so I'm not too familiar with it. My only experience with TDI-related packages is Fast Lisa response, namely for generating EMRIs in the AET channels.

What I wanted to do was perform inference of EMRIs overlaid on a background of glitches to see how much posterior bias occurs. So currently I use Fast Lisa response for this, but it's not clear to me how to include glitches when computing the TDI variables.

-------- Original message --------
From: Jean-Baptiste Bayle ***@***.***>
Date: 17/12/2024 9:16 pm (GMT+12:00)
To: mikekatz04/lisa-on-gpu ***@***.***>
Cc: Amin Boumerdassi ***@***.***>, Author ***@***.***>
Subject: Re: [mikekatz04/lisa-on-gpu] Clarification on "GW projections" (Issue #8)


Can I ask why you don't plan on using, say, PyTDI? Do you plan to do inference?

**Comment by j2bbayle** [comment]
So you won't need to compute the TDI response of these glitches at every step of your sampler?

**Comment by AminBoumerdassi** [comment]
Yes, I think I would need to repeatedly compute the TDI response for the EMRI itself, and only once for the glitch background. Would it be more practical to just use PyTDI for the glitches and then inject the EMRI as needed?

-------- Original message --------
From: Jean-Baptiste Bayle ***@***.***>
Date: 17/12/2024 11:32 pm (GMT+12:00)
To: mikekatz04/lisa-on-gpu ***@***.***>
Cc: Amin Boumerdassi ***@***.***>, Author ***@***.***>
Subject: Re: [mikekatz04/lisa-on-gpu] Clarification on "GW projections" (Issue #8)


So you won't need to compute the TDI response of these glitches at every step of your sampler?

**Comment by j2bbayle** [comment]
You could, but just for consistency (more exactly, to prevent any mismatch in the way you compute your TDI for glitches in EMRI signals), I would indeed use the same tool.

So, back to your question:
> From what I understand, it looks like I should be able to input the projections into the get_tdi_delays() method from the pyresponseTDI class but it's not clear to me what projections Fast LISA response accepts e.g. TMI, ISI etc.

The answer is: the input should be what's equivalent to the eta intermediary variables (which builds a TM-to-TM measurements from the inter-spacecraft (ISI) and test-mass (TMI) interferometric measurements). Equations are as follows

<img width="419" alt="image" src="https://github.com/user-attachments/assets/35b2dd7e-09b9-47c5-bf1a-6d4bb32d7c00" />
<img width="252" alt="image" src="https://github.com/user-attachments/assets/04a8d73b-0619-450d-b6fe-2b123081549b" />

So a glitch in the ISI appears as-is in the eta variables (and you can use this quantity in place of `y_gw` in `get_tdi_delays()`); but you need a bit more work for a TMI glitch.

**Comment by AminBoumerdassi** [comment]
Thanks, I think I'm understanding this now!

I see that ```get_tdi_delays()``` requires up to 6 links, and the eta variables you listed should provide 4 of them (I'm assuming you can just flip the indices for each eta). Is there a complete list of formulae for the eta and xi variables?

And also, I'm wondering if there are any packages that have the delay operators coded up? Ideally, I'd like to avoid having to manually apply the delay operators.

**Comment by j2bbayle** [comment]
Nope, that's 6 etas: 12, 23, 31 and 13, 32, 21. You have the expressions for the first 3 on the first line, and the last 3 on the second line. They are not symmetric so you can't just flip the indices; however, you can use a circular permutation of the first 3 and the last 3, independently. Eg.
```
eta_23 = xi_23 + D_23( rfi_32 - rfi_31) / 2
```

I'm not sure what you mean by "delay operators coded up". I think this package does offer that, as you can define your own TDI combination (see the last section of the tutorial notebook https://github.com/mikekatz04/lisa-on-gpu/blob/master/examples/fast_LISA_response_tutorial.ipynb).

**Comment by AminBoumerdassi** [comment]
Understood about the correct way to handle the indices. A look through the PyTDI documentation helped me write out all the possible etas and xis.

I understand that in the most simple case, an ISI glitch's xi variables reduce to the ISI measurements and that the eta variables reduce to the xi variables. What I'm stuck on is how to handle RFI and TMI glitches. In this case, it's not clear to me how to go from the xi variables to the etas that I would input into Fast Lisa response to get the AET combinations.

Am I correct in thinking that this package doesn't have any functions to calculate the etas based on the ISI, RFI and TMI measurements? My thought process for this is: 

1. input ISI,RFI,TMI measurements into a function that calculates the xi variables
2. Input the xi variables and RFI measurements into a function that calculates the eta variables
3. Input the eta variables into Fast LISA response to get the AET combinations

**Comment by j2bbayle** [comment]
If I'm not mistaking, the way to from RFI and TMI glitches to xi and eta variables is identical to forming a TDI combination such as X, Y, or Z. It's a bunch of delays to be applied, and linearly combining the results. The last section of the tutorial notebook shows how to define such custom combinations (with custom delays); I think you can use that.

**Comment by AminBoumerdassi** [comment]
I'm not sure that the last section of the tutorial is what I'm looking for.

<img width="567" alt="image" src="https://github.com/user-attachments/assets/86fecd2e-835c-4bdd-a117-7caf78fc2480" />

Just to confirm that I've understood the above bit correctly, the custom TDI combinations you can define in this package require all of the etas, the delay operators being applied to each eta, and the sign of each delay operator. The problem is that this setup seems to assume that you've already calculated the etas and that the delay operators act solely on the etas, whereas in my case I only have the etas in terms of interferometer measurements which themselves often require delay operators.

For example, suppose there was an RFI glitch in MOSA 21. I find that:

$\xi_{12} = \frac {D_{12}rfi_{21}}{2}$
$\xi_{23} = \xi_{31} = \xi_{13} = \xi_{32} = 0$
$\xi_{21} = \frac {rfi_{21}}{2}$

So I can rewrite the eta variables as:

$\eta_{12}= D_{12}(rfi_{21})$
$\eta_{23} = \eta_{31} = \eta_{13} = \eta_{21} = \eta_{32} = 0$

Then using the formula for the X1 combination:

<img width="293" alt="image" src="https://github.com/user-attachments/assets/251072da-4363-4cee-acb1-c690945ab85b" />

I can simplify X1 to get:

$X_1 = -(1-D_{131})(\eta_{12}) =  -(1-D_{131})(D_{12} rfi_{21})$

So I can then treat RFI_21 as if it was $\eta_{12}$ and specify all the corresponding delay operators in the custom TDI combination.

I'm not sure how this would work if I wanted to scale this up to a datastream of glitches randomly injected into various interferometers, i.e. that RFI, TMI, ISI measurements all become non-zero over some given window of time. In this case, would it not be more practical to compute the eta variables with PyTDI then input those into Fast Lisa response?

**Comment by j2bbayle** [comment]
I guess you could do this (use PyTDI and then input those into fastlisaresponse) -- but I think you want to avoid the time-intensive CPU TDI calculations.

I think you can work out similar equations (to what you've written) for all RFI and TMI measurements -- ie., get the expression for XYZ in terms of RFI and TMI (12, 23, 31, 13, 32, and 21). Then you can code these up in fastlisaresponse, following the example you screenshot.

**Comment by mikekatz04** [comment]
I am going to close this for now. If you want to continue this discussion, reopen it.
